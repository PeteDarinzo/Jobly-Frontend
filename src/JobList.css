
.JobList {
  position: relative;
  top: 25px;
}

.JobList > div {
  width: 40%;
  margin: 50px auto 0 auto;
}

