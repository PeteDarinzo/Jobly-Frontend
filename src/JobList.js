import React from "react";
import JobCard from "./JobCard";
import "./JobList.css";


/** Generic component to display a job cards */

const JobList = ({ jobs, apply, applications }) => {

  return (
    <div className="JobList">
      <h1>All Jobs:</h1>
      <div>
        {jobs.map(job => (
          <JobCard
            id={job.id}
            title={job.title}
            companyName={job.companyName}
            salary={job.salary}
            equity={job.equity}
            apply={apply}
            applied={applications.indexOf(job.id) >= 0}
            key={job.id}
          />
        ))}
      </div>
    </div>
  );
}

export default JobList;