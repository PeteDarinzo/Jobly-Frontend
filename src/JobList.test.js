import JobList from "./JobList";
import { render, fireEvent } from '@testing-library/react';
import { MemoryRouter } from "react-router-dom"

it("renders a job list without crashing", () => {
  render(
    <MemoryRouter>
      <JobList jobs={[]} apply={() => { }} applications={[]} />
    </MemoryRouter>
  );
});

it("matches the job list snapshot", () => {
  const { asFragment } = render(
    <MemoryRouter>
      <JobList jobs={[]} apply={() => { }} applications={[]} />
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});