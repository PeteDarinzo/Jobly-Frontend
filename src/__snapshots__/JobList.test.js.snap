// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the job list snapshot 1`] = `
<DocumentFragment>
  <div
    class="JobList"
  >
    <h1>
      All Jobs:
    </h1>
    <div />
  </div>
</DocumentFragment>
`;
